### PR TITLE
config_consistency: Enable tests for the python and ruby tracers

### PR DESCRIPTION
**Pull request by mabdinur**
## Motivation

- Enables all python and ruby config consistency tests that are XPassing.

### Changes to server.rb

- Updates the `start_span` endpoint in the ruby parametric app to avoid setting service, resource, and type arguments to an empty string by default. The default values for these arguments MUST be nil.
- Ensures the `find_span` helper in the ruby parametric app can retrieve spans that are not active. There is not need to only allow parametric endpoints to operate on active spans. This change allows us to test scenarios when the tracing api is used AND DD_TRACE_ENABLED=False.

### Minor changes (not required)

- Refactors tests in tests/parametric/test_config_consistency.py to improve readability

## Workflow

1. ⚠️ Create your PR as draft ⚠️
2. Work on you PR until the CI passes (if something not related to your task is failing, you can ignore it)
3. Mark it as ready for review
    * Test logic is modified? -> Get a review from RFC owner. We're working on refining the `codeowners` file quickly.
    * Framework is modified, or non obvious usage of it -> get a review from [R&P team](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X)

:rocket: Once your PR is reviewed, you can merge it!

🛟 [#apm-shared-testing](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X) 🛟

## Reviewer checklist

* [ ] If PR title starts with `[<language>]`, double-check that only `<language>` is impacted by the change
* [ ] No system-tests internal is modified. Otherwise, I have the approval from [R&P team](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X)
* [ ] CI is green, or failing jobs are not related to this change (and you are 100% sure about this statement)
* [ ] A docker base image is modified?
    * [ ] the relevant `build-XXX-image` label is present
* [ ] A scenario is added (or removed)?
    * [ ] Get a review from [R&P team](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X)

### CODE REVIEW

File: manifests/nodejs.yml
Comment (mabdinur) [inline manifests/nodejs.yml:577]:
```suggestion
    # feature should be supported in all recent versions
    Test_Config_UnifiedServiceTagging_CustomService: *ref_3_0_0
```

File: manifests/php.yml
Comment (mabdinur) [inline manifests/php.yml:330]:
```suggestion
    Test_Config_UnifiedServiceTagging_CustomService: v1.0.0 (feature should be supported in all recent versions)
```

File: manifests/cpp.yml
Comment (mabdinur) [inline manifests/cpp.yml:182]:
```suggestion
    Test_Config_UnifiedServiceTagging_CustomService: missing_feature
```

File: manifests/dotnet.yml
Comment (mabdinur) [inline manifests/dotnet.yml:380]:
```suggestion
    Test_Config_UnifiedServiceTagging_CustomService: missing_feature
```

File: manifests/golang.yml
Comment (mabdinur) [inline manifests/golang.yml:497]:
```suggestion
    Test_Config_UnifiedServiceTagging_CustomService: missing_feature
```

File: manifests/java.yml
Comment (mabdinur) [inline manifests/java.yml:1294]:
```suggestion
    Test_Config_UnifiedServiceTagging_CustomService: missing_feature
```

File: manifests/nodejs.yml
Comment (mabdinur) [inline manifests/nodejs.yml:578]:
```suggestion
    Test_Config_UnifiedServiceTagging_CustomService: missing_feature
```

File: manifests/php.yml
Comment (mabdinur) [inline manifests/php.yml:330]:
```suggestion
    Test_Config_UnifiedServiceTagging_CustomService: missing_feature
```

File: tests/parametric/test_config_consistency.py
Comment (mabdinur) [inline tests/parametric/test_config_consistency.py:29]:
```suggestion
            with test_library.start_span("allowed"):
```

File: tests/parametric/test_config_consistency.py
Comment (mabdinur) [inline tests/parametric/test_config_consistency.py:101]:
```suggestion
        assert span1["service"] == "version_test"
        assert span1["meta"]["version"] == "5.2.0"
```

**Comment by mabdinur** [comment]
Failing ruby tests are unreleated

**Review by khanayan123 — APPROVED** [review]

File: manifests/python.yml
Comment (mabdinur) [inline manifests/python.yml:781]:
```suggestion
    Test_Config_ObfuscationQueryStringRegexp_Configured: v2.0.0
```

**Review by cbeauchesne — APPROVED** [review]

File: manifests/python.yml
Comment (mabdinur) [inline manifests/python.yml:785]:
```suggestion
    Test_Config_ObfuscationQueryStringRegexp_Empty: v2.0.0
```

File: manifests/python.yml
Comment (mabdinur) [inline manifests/python.yml:790]:
```suggestion
    Test_Config_ClientTagQueryString_Configured: v2.15.0
```

File: tests/parametric/test_config_consistency.py
Comment (mabdinur) [inline tests/parametric/test_config_consistency.py:159]: Use host and port

File: utils/build/docker/ruby/parametric/server.rb
Comment (mabdinur) [inline utils/build/docker/ruby/parametric/server.rb:601]:
```suggestion
      args.name,
      service: args.service.empty? ? nil : args.service,
      resource: args.resource.empty? ? nil : args.resource,
      type: args.type.empty? ? nil : args.type,
```

**Comment by mabdinur** [comment]
Blocked by: https://github.com/DataDog/dd-trace-rb/pull/4024/files#commit-suggestions. 

Once this PR is merged all ruby tests should be green

File: manifests/python.yml
Comment (mabdinur) [inline manifests/python.yml:793]:
```suggestion
    Test_Config_HttpClientErrorStatuses_FeatureFlagCustom: missing_feature (xpassing but should fail, this environment variable is not supported)
```

File: manifests/python.yml
Comment (mabdinur) [inline manifests/python.yml:797]:
```suggestion
      '*': irrelevant (kafka endpoint is not implemented)
```

File: manifests/python.yml
Comment (mabdinur) [inline manifests/python.yml:800]:
```suggestion
      '*': irrelevant (kafka endpoint is not implemented)
```

File: manifests/ruby.yml
Comment (mabdinur) [inline manifests/ruby.yml:270]:
```suggestion
          rails70: missing_feature (ruby-kafka does not set the kafka topic on spans)
```

File: manifests/ruby.yml
Comment (mabdinur) [inline manifests/ruby.yml:425]:
```suggestion
    Test_Config_ObfuscationQueryStringRegexp_Empty: missing_feature (environment variable is not supported)
```

**Pull request merged**
